xarray/core/common.py

```diff
@@ -1263,11 +1266,27 @@ def where(self, cond, other=dtypes.NA, drop: bool = False):
 
         return ops.where_method(self, cond, other)
 
+    def set_close(self, close: Optional[Callable[[], None]]) -> None:
+        """Register the function that releases all resources used by the data store.
```

Comment:
@shoyer not `set_close` and `close` use slightly different wording:

close -> `Close any files linked to this object`

set_close -> `Register the function that releases all resources used by the data store.`

I would go with `release(s) any resources linked to this object` everywhere, but I'm not a native speaker and I'm generally not very good a writing user documentation.

The important point is that we use the same working everywhere.

Reply:
I like `release(s) any resources linked to this object`